language/stdlib/modules/libra_account.move

```diff
@@ -0,0 +1,492 @@
+    // account_address (i.e., write to account_address/LibraAccount::T/authentication_key).
+    // There is at most one KeyRotationCapability in existence for a given address.
+    resource struct KeyRotationCapability {
+        account_address: address,
+    }
+
+    // Message for sent events
+    struct SentPaymentEvent {
+        // The amount of LibraCoin::T sent
+        amount: u64,
+        // The address that was paid
+        payee: address,
+        // Metadata associated with the payment
+        metadata: bytearray,
+    }
+
+    // Message for received events
+    struct ReceivedPaymentEvent {
+        // The amount of LibraCoin::T received
+        amount: u64,
+        // The address that sent the coin
+        payer: address,
+        // Metadata associated with the payment
+        metadata: bytearray,
+    }
+
+    /// Events
+    // A resource representing the counter used to generate uniqueness under each account. There won't be destructor for
+    // this resource to guarantee the uniqueness of the generated handle.
+    resource struct EventHandleGenerator {
+        // A monotonically increasing counter
+        counter: u64,
+    }
+
+    // A handle for an event such that:
+    // 1. Other modules can emit events to this handle.
+    // 2. Storage can use this handle to prove the total number of events that happened in the past.
+    resource struct EventHandle<T: copyable> {
+        // Total number of events emitted to this event stream.
+        counter: u64,
+        // A globally unique ID for this event stream.
+        guid: bytearray,
+    }
+
+    // Deposits the `to_deposit` coin into the `payee`'s account
+    public fun deposit(payee: address, to_deposit: LibraCoin::T) acquires T {
+        // Since we don't have bytearray literals in the source language at
+        // the moment.
+        // FIXME: Update this once we have bytearray literals
+        deposit_with_metadata(payee, to_deposit, U64Util::u64_to_bytes(0));
+    }
+
+    // Deposits the `to_deposit` coin into the `payee`'s account with the attached `metadata`
+    public fun deposit_with_metadata(
+        payee: address,
+        to_deposit: LibraCoin::T,
+        metadata: bytearray
+    ) acquires T {
+        deposit_with_sender_and_metadata(
+            payee,
+            Transaction::sender(),
+            to_deposit,
+            metadata
+        );
+    }
+
+    // Deposits the `to_deposit` coin into the `payee`'s account with the attached `metadata` and
+    // sender address
+    fun deposit_with_sender_and_metadata(
+        payee: address,
+        sender: address,
+        to_deposit: LibraCoin::T,
+        metadata: bytearray
+    ) acquires T {
+        // Check that the `to_deposit` coin is non-zero
+        let deposit_value = LibraCoin::value(&to_deposit);
+        Transaction::assert(deposit_value > 0, 7);
+
+        // Load the sender's account
+        let sender_account_ref = borrow_global_mut<T>(sender);
```

Comment:
This don’t need to be mut right?

Reply:
It does doesn't it? In particular, we pass in the `sender_account_ref`'s `sent_events` field mutably in to `emit_event` just below this on line 125.

Reply:
Ahhh yes! You are right!